From 4050606d6b90f13b1aa1e4619a019de3e3f2a94b Mon Sep 17 00:00:00 2001
From: Marcelo Vanzin <vanzin@cloudera.com>
Date: Fri, 4 Dec 2015 16:05:51 -0800
Subject: [PATCH 024/201] CLOUDERA-BUILD. Revert "[SPARK-11783][SQL] Fixes
 execution Hive client when using remote Hive
 metastore"

This reverts commit c7f95df5c6d8eb2e6f11cf58b704fea34326a5f2. CDH uses the
same version for execution and metadata; connecting to other metastores is
not supported. And removing this change speeds up the startup of apps that
use HiveContext.

(cherry picked from commit fa708bcd592e76875d2a8077c924e748c5b9cea5)
---
 .../org/apache/spark/sql/hive/HiveContext.scala    |   15 ---------------
 1 file changed, 15 deletions(-)

diff --git a/sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveContext.scala b/sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveContext.scala
index b243db4..ffb629a 100644
--- a/sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveContext.scala
+++ b/sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveContext.scala
@@ -740,21 +740,6 @@ private[hive] object HiveContext {
     propMap.put(HiveConf.ConfVars.SCRATCHDIR.varname, scratchDir.toURI.toString)
     propMap.put("datanucleus.rdbms.datastoreAdapterClassName",
       "org.datanucleus.store.rdbms.adapter.DerbyAdapter")
-
-    // SPARK-11783: When "hive.metastore.uris" is set, the metastore connection mode will be
-    // remote (https://cwiki.apache.org/confluence/display/Hive/AdminManual+MetastoreAdmin
-    // mentions that "If hive.metastore.uris is empty local mode is assumed, remote otherwise").
-    // Remote means that the metastore server is running in its own process.
-    // When the mode is remote, configurations like "javax.jdo.option.ConnectionURL" will not be
-    // used (because they are used by remote metastore server that talks to the database).
-    // Because execution Hive should always connects to a embedded derby metastore.
-    // We have to remove the value of hive.metastore.uris. So, the execution Hive client connects
-    // to the actual embedded derby metastore instead of the remote metastore.
-    // You can search HiveConf.ConfVars.METASTOREURIS in the code of HiveConf (in Hive's repo).
-    // Then, you will find that the local metastore mode is only set to true when
-    // hive.metastore.uris is not set.
-    propMap.put(ConfVars.METASTOREURIS.varname, "")
-
     propMap.toMap
   }
 
-- 
1.7.9.5

